move reporting for Revenue to reportTax

orichters opened the pull request:
To implement a global aggregate of Revenue variables and for greater consistency, move them to reportTax.

This is~~n't~~ beautiful after all, ~~as both reportPrices and reportTax now require input from FE and Emi, which I suspect may slow down the whole process. But as these Emi and FE variables are not necessarily straightforward in their calculation and I'm not really familiar with the different Emi and FE variables in REMIND, and I want to avoid code redundancy and inconsistency, I first implemented it as such. But concrete hints how to improve it warmly welcome.~~

I did that by
- [moving](https://github.com/pik-piam/remind2/pull/227/files#diff-c4e2cb978caee670f1b02a44264a8c7b59215e7b174ed32ad75e5024eae8cba1) `reportTax` further down in the list of `convGDX2MIF` (luckily, the data is not needed beforehand: I took `reportTax` out of the list and it didn't crash as @Renato-Rodrigues suggested)
- in [`reportTax`](https://github.com/pik-piam/remind2/pull/227/files#diff-3ec7318a35d8b5d0412cecd80a6edccb6c52077fecc0fc92cdc93273a3e65950), added a dependence on `output` generated by `reportEmi` and `reportFE` only executed if `output = NULL`. 
- the tests ran through without a problem and also `out[, , "Revenue|Government|Tax|Carbon|+|Demand|Buildings (billion US$2005/yr)"]` looked good. Yeah!

I also
- in [`reportPrices`](https://github.com/pik-piam/remind2/pull/227/files#diff-9079799de4d750220602928460d4d1998c56d82cbb46b1f6b1c95405768aec36), make sure that the calls to `reportEmi` and `reportFE` are done only if `output = NULL`, further speeding up the process
- improved [`compareScenConf.R`](https://github.com/pik-piam/remind2/pull/227/files#diff-e7479f86eb4cd5e10dad5baaa11ac15ef580ab069f272e2e2c21c16758e04854) such that it more consistently does not print scenarios where nothing has changed, they are sorted as in the new file with deleted scenarios at the bottom (Maybe @mikapfl wants to have a look?)
- improved and shortened the `messages`, changing "executing reportFE" to "reportPrices executes reportFE" and merging multiple of these statements in one function to one.
- fixed some of these "join by" messages that populate the log file.

christophbertram reviewed (approved):
Thanks a lot, this looks good, and commit message that explains it well. Great to have the improvements in terms of "joined_by" and "executes..." statements.

orichters commented:
For @mikapfl
```
remind2::compareScenConf(fileList = c("/p/tmp/oliverr/debugging/scenario_config_csv1.csv", "/p/tmp/oliverr/debugging/scenario_config_csv2.csv"))
```
<details>
  <summary>Click to extend old version</summary>

```
File comparison: /p/tmp/oliverr/debugging/scenario_config_csv1.csv -> /p/tmp/oliverr/debugging/scenario_config_csv2.csv
No configfile default.cfg found in /p/tmp/oliverr/remind2 or ../debugging.
Columns deleted: 
Columns added:   description
Renamed columns: 
Renamed rows:    
_____Calibration_____:
Calib_EU:
_____Runs for first years____:
Base:
Ref-gd40:
_____Ariadne scenarios_____:
Trend:
Bal:
Elec_dom:
Elec_imp:
H2_dom:
H2_imp:
Synf_dom:
Synf_imp:
____Tests___:
DEU_Base:
DH was added.
```
</details>
<details>
<summary>Click to extend new version</summary>

```
No configfile default.cfg found in /p/tmp/oliverr/remind2 or ../debugging.

File comparison: ../debugging/scenario_config_csv1.csv -> ../debugging/scenario_config_csv2.csv
Columns deleted: -
Columns added:   description
Renamed columns: -
Renamed rows:    -

Changes in the scenarios:
+ DH was added.
```
</details>

orichters commented on R/reportTax.R:
```diff
@@ -237,6 +244,24 @@ reportTax <- function(gdx,regionSubsetList=NULL,t=c(seq(2005,2060,5),seq(2070,21
+                           + out[, , "Revenue|Government|Tax|Carbon|+|Supply (billion US$2005/yr)"],
+                                         "Revenue|Government|Tax|Carbon (billion US$2005/yr)")
+                  )
+
```
Maybe we can still improve [this part](https://github.com/orichters/remind2/blob/sectoralCO2tax/R/reportTax.R#L253:L269)

mikapfl commented on R/compareScenConf.R:
```diff
@@ -13,15 +13,15 @@
 #'     configfile = "default.cfg",
-#'     renamedColumns = c("old1" = "new1", "old2" = "new2"),
-#'     renamedRows = c("old3" = "new3", "old3" = "new4"))
+#'     renamedCols = c("old1" = "new1", "old2" = "new2"),
+#'     renamedRows = c("old3" = "new3", "old4" = "new4"))
```
Are you sure `renamedColumns` as a function name is not used anywhere? Why change the API here?

orichters replied:
Well, as this function is only some days old, nobody knows about it and `reposearch` founds nothing, so I do not suspect any problems. Why? Just less to type…

mikapfl reviewed (commented):
The compareScenConf changes look good to me. :slightly_smiling_face:
